tutorial6_hierarchical/demo/main.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * CONSTANTS AND GLOBALS
+ * */
+const width = window.innerWidth * 0.9,
+  height = window.innerHeight * 0.7,
+  margin = { top: 20, bottom: 50, left: 60, right: 40 },
+  tooltipSpecs = { width: 100, height: 50 }
+
+let svg;
+let tooltip;
+
+/**
+ * APPLICATION STATE
+ * */
+let state = {
+  data: null,
+  hover: null,
+};
+
+/**
+ * LOAD DATA
+ * */
+d3.json("../../data/flare.json", d3.autotype).then(data => {
+  state.data = data;
+  init();
+});
+
+/**
+ * INITIALIZING FUNCTION
+ * this will be run *one time* when the data finishes loading in
+ * */
+function init() {
+
+  const container = svg = d3
+    .select("#d3-container")
+    .style("position", "relative")
+
+  tooltip = container 
+    .append("div")
+    .attr("width", 100)
+    .attr("height", 100) 
+    .style("position", "absolute")
+    .style("background-color", "white")
```

Comment:
left these in here -- usually I would move to css, but think its worth mentioning in a demo, which prompted me to leave them here. thoughts?